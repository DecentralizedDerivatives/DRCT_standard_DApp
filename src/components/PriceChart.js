import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Highcharts from 'highcharts/highstock';
import api from '../api';
require('highcharts/modules/exporting')(Highcharts);
import highchartsOptions from './highcharts.options';
import '../styles/PriceChart.css';

Highcharts.setOptions({
  rangeSelector: {
    inputEnabled: false,
    align: 'right',
    buttons: [
      {
        type: 'month',
        count: 1,
        text: '1m'
      },
      {
        type: 'all',
        count: 3,
        text: '3m'
      }
    ]
  },

  exporting: {
    enabled: false
  },

  navigator: {
    enabled: false
  },

  colors: [
    '#30AD63' //line
  ],
  chart: {
    backgroundColor: '#EEF2F5',
    style: {
      fontFamily: "'Unica One', sans-serif"
    }
  },
  subtitle: {
    style: {
      color: '#E0E0E3',
      textTransform: 'uppercase'
    }
  },
  xAxis: {
    gridLineWidth: 0,
    minorGridLineWidth: 0,
    tickWidth: 0,
    lineWidth: 0,
    labels: {
      enabled: false
    }
  },
  yAxis: {
    gridLineColor: 'white',
    labels: {
      align: 'left',
      style: {
        color: 'gray'
      }
    }
  },
  scrollbar: {
    enabled: false
  }
});

class PriceChart extends Component {
  constructor() {
    super();
    this.state = {
      currency: 'BTC'
    };
  }

  static currencies = ['BTC', 'ETH'];

  componentDidMount() {
    this.fetchData('btc');
  }

  // TODO:Move into action
  fetchData = async type => {
    const data = await api[type].get();
    this.setState({ data }, () => {
      this.createChart();
    });
  };

  createChart = type => {
    Highcharts.stockChart('container', {
      series: [
        {
          data: this.state.data,
          tooltip: {
            valueDecimals: 2
          },
          enableMouseTracking: false
        }
      ]
    });
  };

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
    this.fetchData(event.target.value.toLowerCase());
  };

  render() {
    return (
      <div className="price-chart">
        <div className="price-chart-item">
          <select onChange={this.handleChange} className="price-chart-dropdown">
            {PriceChart.currencies.map((item, index) => {
              return (
                <option
                  key={'price-chart' + index}
                  className="price-chart-dropdown-item"
                  name={item}
                  value={item}
                >
                  {item}
                </option>
              );
            })}
          </select>
          <div className="price-chart-item">
            <div className="price-chart-item" id="container" />
          </div>
        </div>
      </div>
    );
  }
}

export default PriceChart;
