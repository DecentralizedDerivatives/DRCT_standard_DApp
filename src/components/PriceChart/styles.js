export default theme => ({
  container: {
    width: '400px',
    height: '250px',
  },
  grid: {
    position: 'relative',
    bottom: '27px',
  },
  item: {
    zIndex: 10,
    position: 'relative',
    top: '35px',
    right: '45px',
  },
  dropdown: {
    fontSize: '12px',
  },
  menuItem: {
    padding: '3px',
    fontSize: '11px',
  },
});
