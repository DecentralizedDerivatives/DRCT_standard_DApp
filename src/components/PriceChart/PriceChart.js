import React, { Component } from 'react';
import PropTypes from 'prop-types';
import withStyles from 'material-ui/styles/withStyles';
import Highcharts from 'highcharts/highstock';
import Grid from 'material-ui/Grid';
import Paper from 'material-ui/Paper';
import Dropdown from '../Dropdown';
import theme from './theme';
import styles from './styles';
import api from '../../api';

require('highcharts/modules/exporting')(Highcharts);

// Apply the theme
Highcharts.theme = theme;
Highcharts.setOptions(Highcharts.theme);

class PriceChart extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
  };
  constructor() {
    super();
    this.state = {
      currency: 'BTC',
    };
  }
  static currencies = ['BTC', 'ETH'];

  componentDidMount() {
    this.fetchData('btc');
  }
  fetchData = async type => {
    const data = await api[type].get();
    this.setState({ data }, () => {
      this.createChart();
    });
  };
  createChart = type => {
    Highcharts.stockChart('container', {
      series: [
        {
          data: this.state.data,
          tooltip: {
            valueDecimals: 2,
          },
          enableMouseTracking: false,
        },
      ],
    });
  };

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
    this.fetchData(event.target.value.toLowerCase());
  };

  render() {
    const { classes } = this.props;

    return (
      <Paper className={classes.container}>
        <Grid
          container
          direction="row"
          alignItems="flex-end"
          justify="flex-end"
          className={classes.grid}
        >
          <Grid item className={classes.item}>
            <Dropdown
              menuItems={PriceChart.currencies}
              value={this.state.currency}
              name="currency"
              onChange={this.handleChange}
              className={classes.dropdown}
              menuItemClass={classes.menuItem}
              disableUnderline={true}
              selectBackground="none"
            />
          </Grid>
          <Grid item>
            <div className={classes.container} id="container" />
          </Grid>
        </Grid>
      </Paper>
    );
  }
}

export default withStyles(styles)(PriceChart);
