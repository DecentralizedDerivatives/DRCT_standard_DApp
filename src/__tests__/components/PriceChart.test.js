import mockAxios from 'jest-mock-axios';
import Highcharts from 'highcharts/highstock';

import PriceChart from '../../components/PriceChart';

mockAxios.get.mockImplementation(() => Promise.resolve());

Highcharts.setOptions = jest.fn();
Highcharts.stockChart = jest.fn();

// TODO: increase coverage, setup
describe('<PriceChart />', () => {
  it('renders the component', async () => {
    const wrapper = shallow(<PriceChart store={initFixtureStore()} />).dive();

    wrapper
      .find('.pricechart__dropdown')
      .simulate('change', { target: { value: 'ETH' } });

    expect(Highcharts.setOptions.calls).toMatchSnapshot();
    expect(Highcharts.stockChart.calls).toMatchSnapshot();

    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty component', async () => {
    const wrapper = shallow(<PriceChart store={initStore()} />).dive();

    expect(wrapper).toMatchSnapshot();
  });
});
