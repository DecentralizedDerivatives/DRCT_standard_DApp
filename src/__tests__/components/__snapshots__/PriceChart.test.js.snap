// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceChart /> renders component with error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PriceChart
    getPriceChartData={[Function]}
    pricechart={
      Array [
        Array [
          1533708000000,
          6551.98,
        ],
      ]
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pricechart__item"
      >
        <select
          className="pricechart__dropdown"
          onChange={[Function]}
        >
          <option
            className="pricechart__dropdown-item"
            name="BTC"
            value="BTC"
          >
            BTC
          </option>
          <option
            className="pricechart__dropdown-item"
            name="ETH"
            value="ETH"
          >
            ETH
          </option>
        </select>
        <div
          className="pricechart__item"
        >
          <div
            className="pricechart__item"
            id="container"
          />
        </div>
      </div>,
      "className": "pricechart__chart",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="pricechart__dropdown"
            onChange={[Function]}
          >
            <option
              className="pricechart__dropdown-item"
              name="BTC"
              value="BTC"
            >
              BTC
            </option>
            <option
              className="pricechart__dropdown-item"
              name="ETH"
              value="ETH"
            >
              ETH
            </option>
          </select>,
          <div
            className="pricechart__item"
          >
            <div
              className="pricechart__item"
              id="container"
            />
          </div>,
        ],
        "className": "pricechart__item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                className="pricechart__dropdown-item"
                name="BTC"
                value="BTC"
              >
                BTC
              </option>,
              <option
                className="pricechart__dropdown-item"
                name="ETH"
                value="ETH"
              >
                ETH
              </option>,
            ],
            "className": "pricechart__dropdown",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "price-chart0",
              "nodeType": "host",
              "props": Object {
                "children": "BTC",
                "className": "pricechart__dropdown-item",
                "name": "BTC",
                "value": "BTC",
              },
              "ref": null,
              "rendered": "BTC",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "price-chart1",
              "nodeType": "host",
              "props": Object {
                "children": "ETH",
                "className": "pricechart__dropdown-item",
                "name": "ETH",
                "value": "ETH",
              },
              "ref": null,
              "rendered": "ETH",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="pricechart__item"
              id="container"
            />,
            "className": "pricechart__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pricechart__item",
              "id": "container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pricechart__item"
        >
          <select
            className="pricechart__dropdown"
            onChange={[Function]}
          >
            <option
              className="pricechart__dropdown-item"
              name="BTC"
              value="BTC"
            >
              BTC
            </option>
            <option
              className="pricechart__dropdown-item"
              name="ETH"
              value="ETH"
            >
              ETH
            </option>
          </select>
          <div
            className="pricechart__item"
          >
            <div
              className="pricechart__item"
              id="container"
            />
          </div>
        </div>,
        "className": "pricechart__chart",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              className="pricechart__dropdown"
              onChange={[Function]}
            >
              <option
                className="pricechart__dropdown-item"
                name="BTC"
                value="BTC"
              >
                BTC
              </option>
              <option
                className="pricechart__dropdown-item"
                name="ETH"
                value="ETH"
              >
                ETH
              </option>
            </select>,
            <div
              className="pricechart__item"
            >
              <div
                className="pricechart__item"
                id="container"
              />
            </div>,
          ],
          "className": "pricechart__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  className="pricechart__dropdown-item"
                  name="BTC"
                  value="BTC"
                >
                  BTC
                </option>,
                <option
                  className="pricechart__dropdown-item"
                  name="ETH"
                  value="ETH"
                >
                  ETH
                </option>,
              ],
              "className": "pricechart__dropdown",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "price-chart0",
                "nodeType": "host",
                "props": Object {
                  "children": "BTC",
                  "className": "pricechart__dropdown-item",
                  "name": "BTC",
                  "value": "BTC",
                },
                "ref": null,
                "rendered": "BTC",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "price-chart1",
                "nodeType": "host",
                "props": Object {
                  "children": "ETH",
                  "className": "pricechart__dropdown-item",
                  "name": "ETH",
                  "value": "ETH",
                },
                "ref": null,
                "rendered": "ETH",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="pricechart__item"
                id="container"
              />,
              "className": "pricechart__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "pricechart__item",
                "id": "container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<PriceChart /> renders empty component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PriceChart
    getPriceChartData={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pricechart__item"
      >
        <select
          className="pricechart__dropdown"
          onChange={[Function]}
        >
          <option
            className="pricechart__dropdown-item"
            name="BTC"
            value="BTC"
          >
            BTC
          </option>
          <option
            className="pricechart__dropdown-item"
            name="ETH"
            value="ETH"
          >
            ETH
          </option>
        </select>
        <div
          className="pricechart__item"
        >
          <div
            className="pricechart__item"
            id="container"
          />
        </div>
      </div>,
      "className": "pricechart__chart",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="pricechart__dropdown"
            onChange={[Function]}
          >
            <option
              className="pricechart__dropdown-item"
              name="BTC"
              value="BTC"
            >
              BTC
            </option>
            <option
              className="pricechart__dropdown-item"
              name="ETH"
              value="ETH"
            >
              ETH
            </option>
          </select>,
          <div
            className="pricechart__item"
          >
            <div
              className="pricechart__item"
              id="container"
            />
          </div>,
        ],
        "className": "pricechart__item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                className="pricechart__dropdown-item"
                name="BTC"
                value="BTC"
              >
                BTC
              </option>,
              <option
                className="pricechart__dropdown-item"
                name="ETH"
                value="ETH"
              >
                ETH
              </option>,
            ],
            "className": "pricechart__dropdown",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "price-chart0",
              "nodeType": "host",
              "props": Object {
                "children": "BTC",
                "className": "pricechart__dropdown-item",
                "name": "BTC",
                "value": "BTC",
              },
              "ref": null,
              "rendered": "BTC",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "price-chart1",
              "nodeType": "host",
              "props": Object {
                "children": "ETH",
                "className": "pricechart__dropdown-item",
                "name": "ETH",
                "value": "ETH",
              },
              "ref": null,
              "rendered": "ETH",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="pricechart__item"
              id="container"
            />,
            "className": "pricechart__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pricechart__item",
              "id": "container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pricechart__item"
        >
          <select
            className="pricechart__dropdown"
            onChange={[Function]}
          >
            <option
              className="pricechart__dropdown-item"
              name="BTC"
              value="BTC"
            >
              BTC
            </option>
            <option
              className="pricechart__dropdown-item"
              name="ETH"
              value="ETH"
            >
              ETH
            </option>
          </select>
          <div
            className="pricechart__item"
          >
            <div
              className="pricechart__item"
              id="container"
            />
          </div>
        </div>,
        "className": "pricechart__chart",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              className="pricechart__dropdown"
              onChange={[Function]}
            >
              <option
                className="pricechart__dropdown-item"
                name="BTC"
                value="BTC"
              >
                BTC
              </option>
              <option
                className="pricechart__dropdown-item"
                name="ETH"
                value="ETH"
              >
                ETH
              </option>
            </select>,
            <div
              className="pricechart__item"
            >
              <div
                className="pricechart__item"
                id="container"
              />
            </div>,
          ],
          "className": "pricechart__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  className="pricechart__dropdown-item"
                  name="BTC"
                  value="BTC"
                >
                  BTC
                </option>,
                <option
                  className="pricechart__dropdown-item"
                  name="ETH"
                  value="ETH"
                >
                  ETH
                </option>,
              ],
              "className": "pricechart__dropdown",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "price-chart0",
                "nodeType": "host",
                "props": Object {
                  "children": "BTC",
                  "className": "pricechart__dropdown-item",
                  "name": "BTC",
                  "value": "BTC",
                },
                "ref": null,
                "rendered": "BTC",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "price-chart1",
                "nodeType": "host",
                "props": Object {
                  "children": "ETH",
                  "className": "pricechart__dropdown-item",
                  "name": "ETH",
                  "value": "ETH",
                },
                "ref": null,
                "rendered": "ETH",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="pricechart__item"
                id="container"
              />,
              "className": "pricechart__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "pricechart__item",
                "id": "container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<PriceChart /> renders the component 1`] = `
Array [
  Array [
    Object {
      "chart": Object {
        "backgroundColor": "#EEF2F5",
        "style": Object {
          "fontFamily": "'Unica One', sans-serif",
        },
      },
      "colors": Array [
        "#30AD63",
      ],
      "exporting": Object {
        "enabled": false,
      },
      "navigator": Object {
        "enabled": false,
      },
      "scrollbar": Object {
        "enabled": false,
      },
      "subtitle": Object {
        "style": Object {
          "color": "#E0E0E3",
          "textTransform": "uppercase",
        },
      },
      "xAxis": Object {
        "endOnTick": true,
        "gridLineWidth": 0,
        "labels": Object {
          "enabled": true,
        },
        "lineWidth": 0,
        "minorGridLineWidth": 0,
        "tickWidth": 0,
        "type": "datetime",
      },
      "yAxis": Object {
        "gridLineColor": "white",
        "labels": Object {
          "align": "left",
          "style": Object {
            "color": "gray",
          },
        },
      },
    },
  ],
]
`;

exports[`<PriceChart /> renders the component 2`] = `
Array [
  Array [
    "container",
    Object {
      "rangeSelector": Object {
        "align": "right",
        "allButtonsEnabled": true,
        "buttons": Array [
          Object {
            "count": 60,
            "dataGrouping": Object {
              "forced": true,
              "units": Array [
                Array [
                  "minute",
                  Array [
                    1,
                  ],
                ],
              ],
            },
            "events": Object {
              "click": [Function],
            },
            "text": "1m",
            "type": "minute",
          },
          Object {
            "count": 6,
            "dataGrouping": Object {
              "forced": true,
              "units": Array [
                Array [
                  "minute",
                  Array [
                    30,
                  ],
                ],
              ],
            },
            "events": Object {
              "click": [Function],
            },
            "text": "6h",
            "type": "hour",
          },
          Object {
            "count": 1,
            "dataGrouping": Object {
              "enabled": true,
              "forced": false,
              "units": Array [
                Array [
                  "hour",
                  Array [
                    4,
                  ],
                ],
              ],
            },
            "events": Object {
              "click": [Function],
            },
            "text": "1M",
            "type": "month",
          },
          Object {
            "count": 3,
            "dataGrouping": Object {
              "enabled": true,
              "forced": true,
              "units": Array [
                Array [
                  "day",
                  Array [
                    1,
                  ],
                ],
              ],
            },
            "events": Object {
              "click": [Function],
            },
            "text": "3M",
            "type": "month",
          },
        ],
        "inputEnabled": false,
        "selected": 3,
      },
      "series": Array [
        Object {
          "data": Array [
            Array [
              1533708000000,
              6551.98,
            ],
          ],
          "enableMouseTracking": true,
          "name": "ETH",
          "tooltip": Object {
            "valueDecimals": 2,
          },
        },
      ],
      "tooltip": Object {
        "animation": true,
        "backgroundColor": "#eceeef",
        "shadow": true,
      },
    },
  ],
  Array [
    "container",
    Object {
      "rangeSelector": Object {
        "align": "right",
        "allButtonsEnabled": true,
        "buttons": Array [
          Object {
            "count": 60,
            "dataGrouping": Object {
              "forced": true,
              "units": Array [
                Array [
                  "minute",
                  Array [
                    1,
                  ],
                ],
              ],
            },
            "events": Object {
              "click": [Function],
            },
            "text": "1m",
            "type": "minute",
          },
          Object {
            "count": 6,
            "dataGrouping": Object {
              "forced": true,
              "units": Array [
                Array [
                  "minute",
                  Array [
                    30,
                  ],
                ],
              ],
            },
            "events": Object {
              "click": [Function],
            },
            "text": "6h",
            "type": "hour",
          },
          Object {
            "count": 1,
            "dataGrouping": Object {
              "enabled": true,
              "forced": false,
              "units": Array [
                Array [
                  "hour",
                  Array [
                    4,
                  ],
                ],
              ],
            },
            "events": Object {
              "click": [Function],
            },
            "text": "1M",
            "type": "month",
          },
          Object {
            "count": 3,
            "dataGrouping": Object {
              "enabled": true,
              "forced": true,
              "units": Array [
                Array [
                  "day",
                  Array [
                    1,
                  ],
                ],
              ],
            },
            "events": Object {
              "click": [Function],
            },
            "text": "3M",
            "type": "month",
          },
        ],
        "inputEnabled": false,
        "selected": 3,
      },
      "series": Array [
        Object {
          "data": Array [
            Array [
              1533708000000,
              6551.98,
            ],
          ],
          "enableMouseTracking": true,
          "name": "ETH",
          "tooltip": Object {
            "valueDecimals": 2,
          },
        },
      ],
      "tooltip": Object {
        "animation": true,
        "backgroundColor": "#eceeef",
        "shadow": true,
      },
    },
  ],
]
`;

exports[`<PriceChart /> renders the component 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PriceChart
    getPriceChartData={[Function]}
    pricechart={
      Array [
        Array [
          1533708000000,
          6551.98,
        ],
      ]
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pricechart__item"
      >
        <select
          className="pricechart__dropdown"
          onChange={[Function]}
        >
          <option
            className="pricechart__dropdown-item"
            name="BTC"
            value="BTC"
          >
            BTC
          </option>
          <option
            className="pricechart__dropdown-item"
            name="ETH"
            value="ETH"
          >
            ETH
          </option>
        </select>
        <div
          className="pricechart__item"
        >
          <div
            className="pricechart__item"
            id="container"
          />
        </div>
      </div>,
      "className": "pricechart__chart",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="pricechart__dropdown"
            onChange={[Function]}
          >
            <option
              className="pricechart__dropdown-item"
              name="BTC"
              value="BTC"
            >
              BTC
            </option>
            <option
              className="pricechart__dropdown-item"
              name="ETH"
              value="ETH"
            >
              ETH
            </option>
          </select>,
          <div
            className="pricechart__item"
          >
            <div
              className="pricechart__item"
              id="container"
            />
          </div>,
        ],
        "className": "pricechart__item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                className="pricechart__dropdown-item"
                name="BTC"
                value="BTC"
              >
                BTC
              </option>,
              <option
                className="pricechart__dropdown-item"
                name="ETH"
                value="ETH"
              >
                ETH
              </option>,
            ],
            "className": "pricechart__dropdown",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "price-chart0",
              "nodeType": "host",
              "props": Object {
                "children": "BTC",
                "className": "pricechart__dropdown-item",
                "name": "BTC",
                "value": "BTC",
              },
              "ref": null,
              "rendered": "BTC",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "price-chart1",
              "nodeType": "host",
              "props": Object {
                "children": "ETH",
                "className": "pricechart__dropdown-item",
                "name": "ETH",
                "value": "ETH",
              },
              "ref": null,
              "rendered": "ETH",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="pricechart__item"
              id="container"
            />,
            "className": "pricechart__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pricechart__item",
              "id": "container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pricechart__item"
        >
          <select
            className="pricechart__dropdown"
            onChange={[Function]}
          >
            <option
              className="pricechart__dropdown-item"
              name="BTC"
              value="BTC"
            >
              BTC
            </option>
            <option
              className="pricechart__dropdown-item"
              name="ETH"
              value="ETH"
            >
              ETH
            </option>
          </select>
          <div
            className="pricechart__item"
          >
            <div
              className="pricechart__item"
              id="container"
            />
          </div>
        </div>,
        "className": "pricechart__chart",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              className="pricechart__dropdown"
              onChange={[Function]}
            >
              <option
                className="pricechart__dropdown-item"
                name="BTC"
                value="BTC"
              >
                BTC
              </option>
              <option
                className="pricechart__dropdown-item"
                name="ETH"
                value="ETH"
              >
                ETH
              </option>
            </select>,
            <div
              className="pricechart__item"
            >
              <div
                className="pricechart__item"
                id="container"
              />
            </div>,
          ],
          "className": "pricechart__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  className="pricechart__dropdown-item"
                  name="BTC"
                  value="BTC"
                >
                  BTC
                </option>,
                <option
                  className="pricechart__dropdown-item"
                  name="ETH"
                  value="ETH"
                >
                  ETH
                </option>,
              ],
              "className": "pricechart__dropdown",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "price-chart0",
                "nodeType": "host",
                "props": Object {
                  "children": "BTC",
                  "className": "pricechart__dropdown-item",
                  "name": "BTC",
                  "value": "BTC",
                },
                "ref": null,
                "rendered": "BTC",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "price-chart1",
                "nodeType": "host",
                "props": Object {
                  "children": "ETH",
                  "className": "pricechart__dropdown-item",
                  "name": "ETH",
                  "value": "ETH",
                },
                "ref": null,
                "rendered": "ETH",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="pricechart__item"
                id="container"
              />,
              "className": "pricechart__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "pricechart__item",
                "id": "container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
